android_app/app/src/main/java/com/health/openscale/gui/MainActivity.java

```diff
@@ -715,7 +718,7 @@ public void handleMessage(Message msg) {
                 case RETRIEVE_SCALE_DATA:
                     setBluetoothStatusIcon(R.drawable.ic_bluetooth_connection_success);
                     ScaleMeasurement scaleBtData = (ScaleMeasurement) msg.obj;
-
+                    Toast.makeText(getApplicationContext(), getResources().getString(R.string.info_bluetooth_retrieve_data_successful), Toast.LENGTH_SHORT).show();
```

Comment:
On some scales you will retrieve 100 data point from the scales. Therefore its not good to show everytime a toast message. If a toast message is shown it is handled in `addScaleMeasurement` in `OpenScale.java`. If the parameter silent is true then now toast message is shown. An idea is to implement the vibrate feature to add it after ` runUiToastMsg(infoText);` in `addScaleMeasurement` but beware that it must be run in a ui thread as the  `addScaleMeasurement` will also be triggered during synchronization without a ui.